joern-cli/frontends/c2cpg/src/main/scala/io/joern/c2cpg/astcreation/AstForExpressionsCreator.scala

```diff
@@ -63,66 +82,265 @@ trait AstForExpressionsCreator(implicit withSchemaValidation: ValidationMode) {
+          Some(signature),
+          Some(X2CpgDefines.Any)
         )
-      case _ if rec.root.exists(_.isInstanceOf[NewCall]) =>
-        (DispatchTypes.STATIC_DISPATCH, rec.root.get.asInstanceOf[NewCall].code)
-      case reference: IASTIdExpression =>
-        (DispatchTypes.STATIC_DISPATCH, code(reference))
+
+        createCallAst(callCpgNode, args, base = Some(instanceAst), receiver = Some(instanceAst))
+      case idExpr: CPPASTIdExpression =>
+        val args = call.getArguments.toList.map(a => astForNode(a))
+
+        val name      = idExpr.getName.getLastName.toString
+        val signature = X2CpgDefines.UnresolvedSignature
+        val fullName  = s"${X2CpgDefines.UnresolvedNamespace}.$name:$signature(${args.size})"
+
+        val callCpgNode = callNode(
+          call,
+          code(call),
+          name,
+          fullName,
+          DispatchTypes.STATIC_DISPATCH,
+          Some(signature),
+          Some(X2CpgDefines.Any)
+        )
+
+        createCallAst(callCpgNode, args)
+      case other =>
+        // This could either be a pointer or an operator() call we dont know at this point
+        // but since it is CPP we opt for the later.
+        val args = call.getArguments.toList.map(a => astForNode(a))
+
+        val name      = "<operator>()"
+        val signature = X2CpgDefines.UnresolvedSignature
+        val fullName  = s"${X2CpgDefines.UnresolvedNamespace}.$name:$signature(${args.size})"
+
+        val callCpgNode = callNode(
+          call,
+          code(call),
+          name,
+          fullName,
+          DispatchTypes.STATIC_DISPATCH,
+          Some(signature),
+          Some(X2CpgDefines.Any)
+        )
+
+        val instanceAst = astForExpression(functionNameExpr)
+        createCallAst(callCpgNode, args, base = Some(instanceAst), receiver = Some(instanceAst))
+    }
+  }
+
+  private def astForCCallExpression(call: CASTFunctionCallExpression): Ast = {
+    val functionNameExpr = call.getFunctionNameExpression
+    val typ              = functionNameExpr.getExpressionType
+    typ match {
+      case pointerType: CPointerType =>
+        createPointerCallAst(call, cleanType(ASTTypeUtil.getType(call.getExpressionType)))
+      case functionType: CFunctionType =>
+        functionNameExpr match {
+          case idExpr: CASTIdExpression =>
+            createCFunctionCallAst(call, idExpr, cleanType(ASTTypeUtil.getType(call.getExpressionType)))
+          case _ =>
+            createPointerCallAst(call, cleanType(ASTTypeUtil.getType(call.getExpressionType)))
+        }
       case _ =>
-        (DispatchTypes.STATIC_DISPATCH, "")
+        astForCCallExpressionUntyped(call)
     }
+  }
+
+  private def createCFunctionCallAst(
+    call: CASTFunctionCallExpression,
+    idExpr: CASTIdExpression,
+    callTypeFullName: String
+  ): Ast = {
+    val name      = idExpr.getName.getLastName.toString
+    val signature = ""
```

Comment:
What is the reason for blank string instead of `None`? Is it to simply be explicit in some way that the call signature is "known" to be empty, versus simply empty but might possibly be non-empty?

Reply:
Yes, the signature is conceptionally not `None` but the `""`. It is just that `None` ends up to be represented the same way.